tools/pubsys-config/src/lib.rs

```diff
@@ -49,10 +49,31 @@ impl InfraConfig {
             Ok(Self::default())
         }
     }
+
+    /// Deserializes an InfraConfig from a Infra.lock file at a given path
+    pub fn from_lock_path<P>(path: P) -> Result<Self>
```

Comment:
What makes this function different from `from_path`?

This struct's `impl` might be the best place to de-duplicate the code the loads either `Infra.lock` or `Infra.toml`.

Reply:
Good idea, added another method from_path_or_lock that call from other files
In some cases, we wanted a default to be created in a toml didn't exist either, so I added a default flag -- does that look okay?
`from_lock_path` is different because it uses yaml to deserialize instead of toml